### PR TITLE
Complete and working xunit perf tests.

### PR DESCRIPTION
**Pull request by jonfortescue**

### CODE REVIEW

File: NuGet.Config [hunk 1]
```diff
@@ -13,7 +13,7 @@
   <packageSources>
     <add key="dotnet.myget.org dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" protocolVersion="3" />
     <add key="dotnet.myget.org dotnet-buildtools" value="https://dotnet.myget.org/F/dotnet-buildtools/" />
-    <add key="dotnet.myget.org dotnet-corefxlab" value="https://dotnet.myget.org/F/dotnet-corefxlab/" />
```
Comment (mellinoe) [inline NuGet.Config]: Undo

File: global.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "projects": [ "src" ]
+  "projects": [ "src", "tests" ]
```
Comment (mellinoe) [inline global.json]: I think we can undo this as well, it was only for the BenchmarkDotNet stuff, right?

Comment (jonfortescue) [inline global.json]: ah, that's right! I'll undo both.

**Comment by mellinoe** [comment]
LGTM :+1:

File: tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs [hunk 3]
```diff
@@ -0,0 +1,875 @@
...
+        [InlineData("255")] // Max value
+        [InlineData("0")] // Min value
+        private static void BaselineByteParse(string text)
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                byte value;
+                using (iteration.StartMeasurement())
+                {
+                    Byte.TryParse(text, NumberStyles.None, CultureInfo.InvariantCulture, out value);
```
Comment (KrzysztofCwalina) [inline tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs]: This work is too inexpensive. I worry that the cost of the framework calls is a big par of it. Would it make sense to parse this multiple times per iteration?

File: tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs [hunk 4]
```diff
@@ -0,0 +1,875 @@
...
+        [InlineData("0128")] // leading zero
+        [InlineData("00000000128")] // many leading zeroes
+        [InlineData("-128")] // negative value handling for unsigned types
+        [InlineData("255")] // Max value
+        [InlineData("0")] // Min value
+        private static void ByteArrayToByte(string text)
+        {
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                byte[] utf8ByteArray = UtfEncode(text);
```
Comment (KrzysztofCwalina) [inline tests/System.Text.Primitives.Tests/PrimitivesPerfTests.cs]: You should move encoding out of the loop.

**Comment by KrzysztofCwalina** [comment]
I added some comments in the sources. In addition:
1. Please add instructions on how to run these tests.
2. The main value of the new parasers is that: a) they can parse UTF8, b) the can parse byte[] as opposed to strings, c) they can parse unknown lenght subsections of byte[] buffers, as opposed to having to have exact string representing the parsed type. It would be great to add perf comparisons for the scenarios (above) where the new parsing routines would shine.

**Comment by jonfortescue** [comment]
Wait, for some reason my sln file was changed. Fixing now.

**Comment by KrzysztofCwalina** [comment]
looks good!

**Pull request merged**